subprojects/core/src/main/java/org/gradle/initialization/SettingsFactory.java

```diff
@@ -47,7 +49,7 @@ public SettingsInternal createSettings(GradleInternal gradle, File settingsDir,
                                            Map<String, String> gradleProperties, StartParameter startParameter,
                                            ClassLoaderScope buildRootClassLoaderScope) {
 
-        ClassLoaderScope settingsClassLoaderScope = buildRootClassLoaderScope.createChild("settings");
+        ClassLoaderScope settingsClassLoaderScope = cloneScope(buildRootClassLoaderScope).createChild("settings");
```

Comment:
Would changing BuildSourceBuilder to return a ClassPath instead of a ClassLoaderScope make this easier to handle?

Reply:
I went ahead and changed the implementation to use the advice of introducing `deprecated()` to classloaderscope. Therefore I don't see how changing `BuildSourceBuilder` would make things easier here.

Reply:
I ran the `versionControl` integTests. What I see is the once with buildSrc throw the new deprecation warnings (as I expected.) (Havn't fixed them yet though)